File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day25Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day25Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day25Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day25();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void findMinimumCut() {
        var instance = new Day25();
        var expected = 54;
        var actual = instance.findMinimumCut(lines, 3);
        assertEquals(expected, actual, "The product of the minimum cut " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day25Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day25Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day25Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day25()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testFittingLockKeyPairCount() {
    val expected = 3L
    val actual = Day25().fittingLockKeyPairCount(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The fitting lock/key pair count $actual is not the expected value $expected."
    )
  }

  @Test
  fun testCompleteChronicle() {
    val expected = "Chronicle delivered"
    val actual = Day25().completeChronicle(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The chronicle completion message $actual is not the expected value $expected."
    )
  }
}
